### PR TITLE
WT-13025 Do additional fast paths for NULL or blank configurations

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: src/include/api.h [hunk 1]
```diff
@@ -97,8 +97,13 @@
         bool __set_err = true;                                                              \
         const char *(cfg)[] = {WT_CONFIG_BASE(s, struct_name##_##func_name), config, NULL}; \
         API_SESSION_INIT(s, struct_name, func_name, dh);                                    \
-        if ((config) != NULL)                                                               \
-    WT_ERR(__wt_config_check((s), WT_CONFIG_REF(s, struct_name##_##func_name), (config), 0))
+        if (cfg[1] != NULL) {                                                               \
```
Comment (sueloverso) [inline src/include/api.h:100]: This deserves a comment I think. Maybe something like:
/*
 * Optimize configuration checking. Check for the user sending in an empty configuration string
 * that is known to be in array element 1. If it is empty then set the overall element to NULL to save
 * additional checking.
 * /

File: src/cursor/cur_std.c [hunk 2]
```diff
@@ -970,37 +1003,6 @@ __cursor_reuse_or_init(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const char *
...
-        /*
-         * Dump cursors should not have owners: only the top-level cursor should be wrapped in a
-         * dump cursor.
-         */
-        WT_RET(__wti_curdump_create(cursor, owner, &cdump));
-        *ownerp = cdump;
-        F_CLR(cursor, WT_CURSTD_CACHEABLE);
-    } else
-        cdump = NULL;
     *cdumpp = cdump;
```
Comment (sueloverso) [inline src/cursor/cur_std.c:1006]: While I know this is out of the conditional that you moved, it feels weird to leave this line now far away from the dump related code. Can you consider moving it up too?

**Review by sueloverso — APPROVED** [review]
LGTM. I have two minor comments that you can decide if you want.

File: src/cursor/cur_std.c (hunk 2 above)
Comment (ddanderson) [reply src/cursor/cur_std.c]: Yes, moved up closer.

File: src/include/api.h (hunk 1 above)
Comment (ddanderson) [reply src/include/api.h]: I added a comment, slightly different than your wording.

**Comment by ddanderson** [comment]
@tod-johnson-mongodb, do you have time to do a review on this one?

**Pull request merged**